### ISSUE TITLE
Space character after an emoji character is now very long (Avalonia 11)

### ISSUE THREAD

**Issue by Skyedra** [AvaloniaUI/Avalonia#14011]
## Describe the bug
Space displayed after an emoji is very long in Avalonia 11.  This is true for various elements, such as text input entry and labels, etc

## To Reproduce

1. Clone https://github.com/AvaloniaUI/Avalonia.Samples
2. Update MasicMvvmSample.csproj to use dotnet8 (also reproduceable in dotnet 7) and latest Avalonia (currently `11.0.6`)
4. cd into MVVM/BasicMvvmSample and dotnet run
5. In textbox, enter an emoji (such as `🐲` (the best emoji))
6. If you enter a space after the emoji, it is weirdly long

## Expected behavior
Spaces are consistent size, like in Avalonia 10

## Screenshots

https://github.com/AvaloniaUI/Avalonia/assets/22365940/625a8360-059b-4034-9f7b-68e89a67dd01

## Environment 

- OS: Linux Ubuntu 23.04
- Avalonia-Version: 11.0.6

**Comment by timunie** [comment]
can you try nightly pls? https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed

**Comment by Skyedra** [comment]
FYI tested on just released Avalonia 11.0.7 and still an issue (now on Ubuntu 23.10)

**Comment by timunie** [comment]
@Skyedra nightly has still fixes in that 11.0.7 hasn't and thus will only be available in 11.1.

**Comment by Skyedra** [comment]
Ahhhh, ok thanks

**Comment by Skyedra** [comment]
I placed the `nuget.config` file from that article in the root project folder and did a dotnet restore, but I still see the issue.  Do I need to set the version in the csproj to something specific, and if so, to what?

**Comment by timunie** [comment]
https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed

**Comment by Skyedra** [comment]
Yeah, I saw that, but that article assumes I already have an intermediate knowledge of nuget (which I definitely do not :sweat_smile: )

Please document it more clearly on how a CLI user can pull nightly and I will try to follow it.

**Comment by timunie** [comment]
> but that article assumes I already have an intermediate knowledge of nuget

in that case please read nuget docs pls or wait for 11.1 

Other idea: If you want to, you can also run src / samples / ControlCatalog. Read build instructions. https://github.com/AvaloniaUI/Avalonia/blob/master/Documentation/build.md

**Comment by 3aIno** [comment]
Confirming that this is still an issue on 11.2.999-cibuild0047060-alpha.

<img width="273" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/121347511/5af20b2f-c378-4320-b632-42cdd1ff693d">


Emojis also render smaller on my Macbook Pro inbuilt screen compared to my external monitor.

<img width="266" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/121347511/fae77799-c754-48c8-bb50-bb319dbbaef6">

**Comment by jpgarza93** [comment]
Still seeing this issue

**Comment by JohnCampionJr** [comment]
Still reproduces on **12.1.1** (macOS 15, arm64). Here is a root-cause measurement, since the thread doesn't have one yet.

### Measurements

Measured with `FormattedText` at `FontSize=16`, isolating the space by differencing (`W("🙂 b") - W("🙂b")`):

| | space advance |
|---|---|
| plain space | 4.45 |
| space **immediately after** an emoji | **16.00** (3.6×) |
| space later in the same line | 4.45 (normal) |

Only the space *adjacent* to the emoji is affected; a second space is normal. 16.00 is exactly 1em, which is the emoji font's own space glyph.

### Cause

Itemization switches to the fallback (emoji) typeface for the emoji, then keeps consuming characters **while that typeface still has a glyph for them**. Every emoji font ships U+0020, so the following space is absorbed into the emoji run and shaped with the emoji font instead of reverting to the primary one.

Two things confirm it:

**It is independent of the primary font.** The wide space is 16.00 whether the primary is the system default, an embedded TTF, or a composite `#Family, $Default`. Substituting a *different* emoji font changes the number — registering Noto Emoji via `FontManagerOptions.FontFallbacks` gives 20.31 instead of 16.00 — which is what you'd expect if the advance is coming from the emoji font's space glyph.

**Characters the emoji font lacks break the run correctly.** Same test with a separator between the emoji and the space:

| separator after the emoji | in emoji-font cmap? | resulting space |
|---|---|---|
| (none) | U+0020 present | 16.00 |
| U+00A0 no-break space | absent | **4.45** |
| U+2060 word joiner | absent | **4.45** |
| U+200B zero-width space | absent | **4.45** |
| U+FEFF | absent | **4.45** |

cmap checked directly on `NotoColorEmoji.ttf` and `NotoEmoji-Regular.ttf`: both carry U+0020 and none of the other four. So the run-extension rule is behaving as written — U+0020 is simply the one whitespace character emoji fonts do provide, which makes it the one case where the run should end but doesn't.

### Repro

```csharp
var tf = new Typeface(FontFamily.Default);
double W(string s) => new FormattedText(s, CultureInfo.InvariantCulture,
    FlowDirection.LeftToRight, tf, 16, Brushes.Black).Width;

Console.WriteLine(W("a b")      - W("ab"));       // 4.45
Console.WriteLine(W("\U0001F642 b") - W("\U0001F642b")); // 16.00
```

No application-side workaround exists that doesn't alter the text: the only lever is inserting a character the user did not type, which then travels into anything the text is copied or submitted into.
